src/magic_folder/snapshot.py

```diff
@@ -164,6 +164,58 @@ def _get_synchronous_content(self):
         with open(self.content_path, "rb") as f:
             return f.read()
 
+    def serialize(self):
+        """
+        Serialize the LocalSnapshot to JSON.
+
+        :returns: A JSON string representation of the LocalSnapshot
+        """
+        # Recursively serialize into one object.
+
+        def _serialized_dict(local_snapshot):
+            serialized = {
+                'name' : local_snapshot.name,
+                'metadata' : local_snapshot.metadata,
+                'content_path' : local_snapshot.content_path,
+                'parents_local' : [ _serialized_dict(parent) for parent in local_snapshot.parents_local ]
+            }
+
+            return serialized
+
+        serialized = _serialized_dict(self)
+
+        return (json.dumps(serialized))
+
+    @classmethod
+    def deserialize(cls, serialized, author):
+        """
+        Creates a LocalSnapshot from a serialized string that represents the LocalSnapshot.
+
+        :param str serialized: the JSON string that represents the LocalSnapshot
+
+        :param author: an instance of LocalAuthor
+
+        :returns: A LocalSnapshot object representation of the JSON serialized string.
+        """
+        local_snapshot_dict = json.loads(serialized)
+
+        def deserialize_dict(snapshot_dict, author):
+            name = snapshot_dict["name"]
+
+            # deserialize metadata
+            metadata = {}
+            metadata["ctime"] = snapshot_dict["metadata"]["ctime"]
+            metadata["mtime"] = snapshot_dict["metadata"]["mtime"]
+
```

Comment:
Above, we serialized everything that's in metadata; why are we only looking at these two now? In Tahoe it seems that the metadata is pretty open-ended so I think we should probably just say `metadata = snapshot_dict["metadata"]` to take all of it.